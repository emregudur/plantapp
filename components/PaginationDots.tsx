import React from 'react'
import { StyleSheet, View } from 'react-native'

interface Props {
  activeIndex: number
}

export default function SliderDots({ activeIndex }: Props) {
  return (
    <View style={styles.dotsContainer}>
      {Array(3)
        .fill('')
        .map((_, i) => {
          return <View key={i} style={i === activeIndex ? styles.dotActive : styles.dot}></View>
        })}
    </View>
  )
}

const styles = StyleSheet.create({
  dotsContainer: {
    flex: 1,
    alignContent: 'center',
    alignItems: 'center',
    gap: 12,
    justifyContent: 'center',
    flexDirection: 'row',
  },
  dotActive: {
    width: 10,
    height: 10,
    borderWidth: 0,
    borderRadius: 50,
    backgroundColor: '#13231B',
  },
  dot: {
    width: 6,
    height: 6,
    borderWidth: 0,
    borderRadius: 50,
    backgroundColor: '#13231B40',
  },
})
