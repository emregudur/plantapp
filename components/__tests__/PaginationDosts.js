import * as React from 'react'
import { render } from '@testing-library/react-native'
import PaginationDots from '../PaginationDots'

it(`renders correctly`, () => {
  const tree = render(<PaginationDots activeIndex={0} />).toJSON()

  expect(tree).toMatchSnapshot()
})

it(`should render active index correctly`, async () => {
  const { getByTestId } = render(<PaginationDots activeIndex={0} />)

  expect(getByTestId('pagination-dots').children).toHaveLength(3)
})

it(`should active dots correctly`, async () => {
  let index = 2
  const { getByTestId } = render(<PaginationDots activeIndex={index} />)

  expect(getByTestId('pagination-dots').children[2].props.style.width).toBe(10)
})
