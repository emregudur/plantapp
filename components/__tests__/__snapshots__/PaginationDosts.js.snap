// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "gap": 12,
      "justifyContent": "center",
    }
  }
  testID="pagination-dots"
>
  <View
    style={
      {
        "backgroundColor": "#13231B",
        "borderRadius": 50,
        "borderWidth": 0,
        "height": 10,
        "width": 10,
      }
    }
  />
  <View
    style={
      {
        "backgroundColor": "#13231B40",
        "borderRadius": 50,
        "borderWidth": 0,
        "height": 6,
        "width": 6,
      }
    }
  />
  <View
    style={
      {
        "backgroundColor": "#13231B40",
        "borderRadius": 50,
        "borderWidth": 0,
        "height": 6,
        "width": 6,
      }
    }
  />
</View>
`;
